mobsf/templates/static_analysis/android_binary_analysis.html

```diff
@@ -2145,28 +2141,13 @@ <h5 class="description-header">{{ code_analysis.summary.suppressed }}</h5>
                <div class="list-group">
               {% if app_type not in 'so' %}
                   <p><strong>From APK Resource</strong></p>
-                  <p>  
-                    {% for key, val in strings.items %}
-                      {% if key == 'strings_apk_res' %}
-                        {% for v in val %}
-                          {{ v }}<br/>
-                        {% endfor %}
-                      {% endif %}
-                    {% endfor %}                  
-                  </p>
-                   <p><strong>From Code</strong></p>
-                  <p>  
-                    {% for key, val in strings.items %}
-                      {% if key == 'strings_code' %}
-                        {% for v in val %}
-                          {{ v }}<br/>
-                        {% endfor %}
-                      {% endif %}
-                    {% endfor %}                  
-                  </p>
+                  {% include 'base/list.html' with list=strings.strings_apk_res type="strings" %}
+                  <p><strong>From Code</strong></p>
+                  {% include 'base/list.html' with list=strings.strings_code type="strings" %}
               {% endif %}
                     <p><strong>From Shared Objects</strong></p>
-                  <p>  
+                    {% include 'base/list.html' with list=strings.strings_so type="strings" %}
```

Comment:
The shared object strings are separated by filename. The datastructure is a dict of list. For example

```
{'apktool_out/lib/armeabi-v7a/librealm-jni.so': ['', '_ZNSt6__ndk113basic_istreamIcNS_11char_traitsIcEEErsEPFRNS_8ios_baseES5_E', 'hJk8', 'OCSP_REQINFO_free', '$@`@?$', 'no content type', 'aria-192-ecb', 'Unsupported column type', 'G FQF', '3I(FM"yDd', '-tIxDyD', 'vms_load', 'LI "1\'@', '_ZNSt6__ndk113basic_ostreamIwNS'], 'foo.so': ['string', 'string2']}
```

So here it gets rendered incorrectly.


<img width="1418" alt="Screenshot 2024-03-06 at 9 29 54 PM" src="https://github.com/MobSF/Mobile-Security-Framework-MobSF/assets/4301109/ae0f52f4-cb4e-4e2f-8943-572107b70856">

Instead of the number of strings, it shows the no of shared object files.

Reply:
I got some time to work on refactoring this.